provider/datagen/src/registry.rs

```diff
@@ -81,6 +81,23 @@ macro_rules! registry {
             )+
             unreachable!("unregistered marker")
         }
+
+        #[doc(hidden)]
+        pub fn deserialize_and_discard<R>(key: DataKey, buf: DataPayload<BufferMarker>, r: impl Fn() -> R) -> Result<R, DataError> {
```

Comment:
question: is there a reason we need the callback? It's just called after deserialization, no? We'll know deserilization was successful from the error value of this function anyway

Reply:
There's a comment at the call site. We need to get the heap stats before the function returns, because otherwise the struct is dropped and we cannot distinguish between net violations and total violations.